test/modern_bpf/event_class/event_class.h

```diff
@@ -0,0 +1,335 @@
+	 *
+	 */
+	~event_test();
+
+	/**
+	 * @brief Mark only the 64-bit syscall with `syscall_id` as interesting.
+	 *
+	 * @param syscall_id id of the syscall.
+	 */
+	void mark_single_64bit_syscall_as_interesting(int syscall_id);
+
+	/**
+	 * @brief Mark all 64-bit syscalls as uninteresting.
+	 *
+	 */
+	void mark_all_64bit_syscalls_as_uninteresting();
+
+	/**
+	 * @brief Tracepoints can start to catch events.
+	 *
+	 */
+	void enable_capture();
+
+	/**
+	 * @brief Deny the Tracepoints to catch further events.
+	 *
+	 */
+	void disable_capture();
+
+	/**
+	 * @brief Clear the ring buffers from all previous events until they
+	 * are all empty.
+	 *
+	 */
+	void clear_ring_buffers();
+
+	/**
+	 * @brief Parse information from the event that we have extracted from the buffer:
+	 * - Number of parameters.
+	 * - Lenght and value of each parameter.
+	 * - Total length of the event.
+	 *
+	 */
+	void parse_event();
+
+	/////////////////////////////////
+	// GENERIC EVENT ASSERTIONS
+	/////////////////////////////////
+
+	/**
+	 * @brief Assert if in our buffers there is an event of the specified type.
+	 * Search until all buffers are empty. If we don't find any event of this type
+	 * we fail.
+	 */
+	void assert_event_presence();
+
+	/**
+	 * @brief Assert some fields of the event header:
+	 * - the event params num must match the number of parameters in the event table.
+	 * - the overall event length must match the total length written in the header.
+	 */
+	void assert_header();
+
+	/**
+	 * @brief Assert the number of params that bpf-side pushes to userspace
+	 *  against the number of params written in the BPF table.
+	 *
+	 * @param total_params total number of params we send bpf-side
+	 */
+	void assert_num_params_pushed(int param_num);
+
+	/////////////////////////////////
+	// SYSCALL RESULT ASSERTIONS
+	/////////////////////////////////
+
+	/**
+	 * @brief When we call this function we expect that the syscall will fail.
+	 * Please note that not all syscalls return `-1` when they fail, there are some
+	 * exceptions... by the way, with this function we can manage the large majority of
+	 * cases. When you use this method you must check what is the syscall return value!
+	 *
+	 * @param syscall_rc the return code of the syscall to assert
+	 * @param syscall_name the name of the syscall to assert
+	 */
+	void assert_syscall_failure(long syscall_rc, const char* syscall_name);
+
+	/**
+	 * @brief When we call this function we expect that the syscall will succeed.
+	 * Please note that not all syscalls return `-1` when they fail, there are some
+	 * exceptions... by the way, with this function we can manage the large majority of
+	 * cases. When you use this method you must check what is the syscall return value!
+	 *
+	 * @param syscall_rc the return code of the syscall to assert
+	 * @param syscall_name the name of the syscall to assert
+	 */
+	void assert_syscall_success(long syscall_rc, const char* syscall_name);
```

Comment:
Do we want to add a third parameter to these assertions? Something like:
```suggestion
	void assert_syscall_success(long syscall_rc, const char* syscall_name, long expected_rc = -1);
```
That way we can easily change the expected error when it is not -1